hack/make-rules/image.sh

```diff
@@ -90,6 +90,9 @@ function build_images() {
     set -x
     docker build --build-arg GO_LDFLAGS="${GO_LDFLAGS}" -t kubeedge/${IMAGE_NAME}:${IMAGE_TAG} -f ${DOCKERFILE_PATH} .
     set +x
+
+    docker builder prune -f
+    docker system prune -f
```

Comment:
```suggestion
    if [[ "${CLEAN_BUILD_IMAGE_CACHE}" = "true" ]]; then
    docker builder prune -f
    docker system prune -f
fi
    
```

At L43, CLEAN_BUILD_IMAGE_CACHE=${CLEAN_BUILD_IMAGE_CACHE:-"false"}

How about add a flag to control whether to clean the cache, because in develpers' host, it's no need and should not delete other images. We only need to delete in ci host now   cc @WillardHu @Shelley-BaoYue